### PR TITLE
[READY] Giving circuits possibility to load item while in assembly

### PR DESCRIPTION
**Pull request by Shdorsh**
[Changelogs]: #Adds the demand_object_input var and change the attackby() script so certain circuits can get an item while they are in an assembly. Also changed the attack_self proc so it doesn't require 2 lists and a list.Find() proc. It only uses 1 list now and that should do the trick. Also fixed a bug where the battery was inserted in the assembly but rather showed up on the location of the assembly instead of going into its contents.

:cl: Shdorsh
tweak:  Makes it possible to create circuits that can get an item loaded into them while they are in an assembly and the assembly is open.
code:  Optimized electronic assemblies also.
fix:  A bug pertaining putting batteries in assemblies
/:cl:

[why]: # Initiating 2 lists is unneeded, so I optimized it all a little bit. Also, makes it easier to craft circuits.

### CODE REVIEW

**Comment by Cyprex** [comment]
The part between CL is player-facing, the part above is only for github users, it seems you need to swap them.

**Comment by Garen7** [comment]
great change, there are a lot of circuits that need something like this

**Comment by Shdorsh** [comment]
thanks man

**Comment by BlueNothing** [comment]
We've kind of needed this for a while.

**Comment by PKPenguin321** [comment]
Very nice PR, makes loading circuits that require specific items (gun firers, grenade primers) less hacky. Doesn't actually buff or nerf circuits, just straight up QoL.

**Comment by Shdorsh** [comment]
What I experienced is that whenever your assembly is in your backpack, adding a battery to it will add the battery in the backpack and the assembly, so you can take the battery from the backpack and put it in other assemblies to link it all to the same battery. This might be exploited to create energizing assemblies made of pure photovoltaic cells or tesla relays that keep energizing the battery while the real assembly can just walk off and do its job far away from any apcs it could drain and lightsources.

**Comment by BlueNothing** [comment]
Or just to run a bunch of assemblies off of one battery, saving time
charging more.

**Comment by BlueNothing** [comment]
Or, just put one of the networked batteries on a charger.

**Comment by Shdorsh** [comment]
Build success, yet errored, no idea what's going on. My own modded version on my desktop works without any porblems though...

**Comment by optimumtact** [comment]
another pr lost to maintainer disinterest

**Comment by Shdorsh** [comment]
Whelp nevermind, I'll port it to hippie, we got the same taste in circuits shenanigans.

**Comment by PKPenguin321** [comment]
oranges reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeopen this NOW

**Comment by optimumtact** [comment]
it's not magically going to make someone care enough to merge it so no, go find a maintainer who wants to merge it and ask them.

**Comment by PKPenguin321** [comment]
😡

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

File: code/modules/integrated_electronics/core/assemblies.dm [hunk 1]
```diff
@@ -491,30 +518,33 @@
...
+		if(input_selection.len ==1)
+			choice = input_selection[input_selection[1]]
 		else
 			var/selection = input(user, "What do you want to interact with?", "Interaction") as null|anything in input_selection
 			if(!check_interactivity(user))
 				return
 			if(selection)
-				var/index = input_selection.Find(selection)
-				choice = available_inputs[index]
+				choice = input_selection.[selection]
```
[the remark this answers is not in the document]
Comment (Shdorsh) [reply code/modules/integrated_electronics/core/assemblies.dm]: wow, I must have been drunk while I did this. amazing what you can find out with a bit of strawberry juice, amaretto and concentrated lemon juice.

**Pull request merged**